pkg/deploy/reaper/reaper.go

```diff
@@ -30,6 +32,27 @@ type DeploymentConfigReaper struct {
 // zero replicas, waits for all of them to get deleted and then deletes both the
 // replication controller and its deployment configuration.
 func (reaper *DeploymentConfigReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) error {
+	// Pause the deployment config marked for deletion first.
+	deployment, err := reaper.updateDeploymentWithRetries(namespace, name, func(d *deployapi.DeploymentConfig) {
+		d.Spec.RevisionHistoryLimit = util.Int32Ptr(0)
+		d.Spec.Replicas = 0
+		d.Spec.Paused = true
+	})
+	if err != nil {
+		return err
+	}
+
+	// Determine if the deployment config controller noticed the pause.
+	if err := wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+		dc, err := reaper.oc.DeploymentConfigs(namespace).Get(name)
+		if err != nil {
+			return false, err
+		}
+		return dc.Status.ObservedGeneration >= deployment.Generation
```

Comment:
You need to keep the generation from the deployment config you got above and use that instead of using the generation of the object you get every time.